extension/src/state_aggregate.rs

```diff
@@ -0,0 +1,464 @@
+//! SELECT duration_in('STOPPED', states) as run_time, duration_in('ERROR', states) as error_time FROM (
+//!   SELECT state_agg(time, state) as states FROM ...
+//! );
+//!
+//! Currently requires loading all data into memory in order to sort it by time.
+
+// TODO How to mark everything experimental?  I can't seem to get these 3:
+// unexpectedly released features: [
+//     "function stateagg_in(cstring)",
+//     "function stateagg_out(stateagg)",
+//     "type stateagg",
+// ]
```

Comment:
for these you put the type definition and `ron_inout_funcs!` in a `#[pg_schema]` module [like so](https://github.com/timescale/timescaledb-toolkit/blob/1c98deac2121402962e6284b9d5534c780d15c38/extension/src/topn.rs#L37-L54)